### PR TITLE
Fixed calculation of oSettings._iDisplayStart in patternfly-functions.js

### PR DESCRIPTION
**Pull request by ameron83**
## Description
Fix for DataTables bootstrap_input pagination controller when the table includes exactly one row.

Relates to https://github.com/patternfly/patternfly/issues/555

## Changes
Changed calculation of oSettings._iDisplayStart when iNewStart is greater than oSettings.fnRecordsDisplay()

### CODE REVIEW

**Comment by bleathem** [comment]
Thanks for the PR @ameron83.  Have you confirmed that the change you are suggesting doesn't break any existing functionality?  If you could share a code sample we could use to reproduce and verify this fix, that would help us in evaluating this PR.

**Comment by ameron83** [comment]
As far as I can tell, no existing functionality is broken with this fix.
Here is an example ReactJS component that displays datatables. The data now includes 2 items, therefore there is no issue. If you remove one item, the error will show up.

`class ItemsTable extends React.Component {
	constructor(props) {
		super(props);
		this.table = null;
	}

	componentDidMount() {
		this.initTable();
	}

	get tableDOM() {
		return $(ReactDOM.findDOMNode(this.refs.table));
	}

	initTable() {
		this.table = this.tableDOM.DataTable({
			autoWidth: false,
			info: true,
			lengthChange: false,
			data: [{c1: 5, c2: 6}, {c1: 1, c2: 2}],
			language: {
				zeroRecords: ReactDOMServer.renderToStaticMarkup(
					<div className="text-center">
						No matching records found.
					</div>
				),
			},
			dom:
				"<'dataTables_header' <'text-right' i > >" +
				"<'table-responsive' t >" +
				"<'dataTables_footer' p >",
			columns: [
				{
					title: 'Col 1',
					name: 'col1',
					data: 'c1',
					className: 'text-right',
					width: '50%',
				},
				{
					title: 'Col 2',
					name: 'col2',
					data: 'c2',
					className: 'text-right',
					width: '50%',
				},
			],
		});
	}

	render() {
		return (
			<div>
				<table
					ref="table"
					className="datatable table table-striped table-bordered table-hover"
				/>
			</div>
		);
	}
}`

**Comment by jmikulka** [comment]
Hi @ameron83, I've tested your fix on your example and found out that the error is still thrown if you enter anything less than 1.

@bleathem, the original implementation (without @ameron83 fix) throws the error if you enter anything less than 1 or anything greater than the last page.

My working example:
```
import $ from 'jquery';
import React from 'react';

export class Table extends React.Component {
	componentDidMount() {
		$(this.table).DataTable({
			data: [
				{c1: 1, c2: 2},
				{c1: 3, c2: 4},
			],
			columns: [
				{ title: 'c1', name: 'c1', data: 'c1' },
				{ title: 'c2', name: 'c2', data: 'c2' },
			],
		});
	}

	render() {
		return (
			<table
				ref={(table) => { this.table = table; }}
				className="datatable table table-striped table-bordered table-hover"
			/>
		);
	}
}
```

**Comment by ameron83** [comment]
@jmikulka Thanks for pointing out the cases in which an error would still be triggered. I committed a fix that takes care of them

File: src/js/patternfly-functions.js [hunk 1]
```diff
@@ -171,9 +171,9 @@
             }
 
             iNewStart = oSettings._iDisplayLength * (this.value - 1);
-            if (iNewStart >= oSettings.fnRecordsDisplay()) {
+            if (iNewStart >= oSettings.fnRecordsDisplay() || iNewStart < 0) {
               /* Display overrun */
-              oSettings._iDisplayStart = (Math.ceil((oSettings.fnRecordsDisplay() - 1) /
+              oSettings._iDisplayStart = (Math.ceil(oSettings.fnRecordsDisplay() /
```
Comment (dlabrecq) [inline src/js/patternfly-functions.js:176]: Downloaded the source and tested this solution against our test page. That is, after modifying the table to have only one row.

I see the JS error on Chrome, but Firefox doesn't appear to generate any errors.

Oddly, pagination jumps to the last page when a negative value is entered. Although, this doesn't appear to be related to these changes -- seems like a DataTables quirk to me.

That said, I don't believe folks will often enter negative values, so I'm good with merging. Much better than a JS error.

**Review by dlabrecq — APPROVED** [review]
LGTM

**Pull request merged**